### DISCUSSION TITLE
Add charge_current_request and charge_current_request_max to MQTT topics

### DISCUSSION THREAD

**Discussion by pkuehnel** [teslamate-org/teslamate#2412] [Ideas]
I created a service to adjust multiple car's current based on PV production in combination with [TeslaMateApi](https://github.com/tobiasehlert/teslamateapi). I have an issue when the car does not use all the power it requests (e.g. battery nearly at 100%, or car is only preconditioning but not charging). It would help me a lot if you could add `charge_current_request` and `charge_current_request_max` to the MQTT topics. Call to Tesla API is `https://owner-api.teslamotors.com/api/1/vehicles/vehivleId/data_request/charge_state`. I am not that deep into Elixir, so I can not do that on my own. Anyone here who can create a PR?

**Comment by pkuehnel** [comment]
@adriankumpf My software already exists (in very early stages: [pkuehnel/SmartTeslaAmpSetter](https://github.com/pkuehnel/SmartTeslaAmpSetter)) but it is really critical for it that there are MQTT topics for `charge_current_request` and `charge_current_request_max`. Can you please add these two?

**Comment by adriankumpf** [comment]
Hi @pkuehnel, I've added those two topics in a8bcb906b1ed43a6f43f538a5d2a509265e4d48d.

**Comment by migube** [comment]
HI guys
I'm currently running both a smart meter setup, gathered in Home assistant, and teslamate in docker.
Do I understand that the charge_current_request MQTT topic will allow to read the values, but Teslamate will not allow (yet?) to write/change the charging_amps paramater?
There is an home assistant custom integration https://community.home-assistant.io/t/optimizing-use-of-unused-solar-power-to-charge-a-tesla/340818 that I could use, but I'd prefer to have it all via teslamate.
Potentially, with pkuehnel extra sw ?

Any thoughts how to achieve the same results from home assistant easily via Teslamate (and if yes how/by when? :-) 
tx!

**Reply by pkuehnel** [reply]
You can use my software if you want to charge solar power, or https://github.com/tobiasehlert/teslamateapi if you just want so change current via API request.

**Reply by migube** [reply]
hi pkuehnel

ok so just installed it, and installed postman to generate some test POST URl's

it looks like most without parameters do work, do I have issues with parameters

e.g opening the trunk keeps saying i didn't provide the parameter

while e.g. command/set_charge_limit?percent=75
{
    "response": {
        "reason": "",
        "result": true
    }
}

but eventually it sets it to 50%  ?! (same with other values...)

do u have some postman examples to be sure it's not me?
tx already!

**Reply by pkuehnel** [reply]
![Screenshot_20220720-154040](https://user-images.githubusercontent.com/35361981/179997243-2bb4ec71-1f09-4073-92ab-8e098e8d124f.png)
As I am on my phone, I only can send you a screenshot. You have to send the variables as serialized Json in your request body.

**Comment by migube** [comment]
that seems to do the trick !
tx again, will then star playing with Home assistant Json tests :) & let u know
merci !
